src/nvim/globals.h

```diff
@@ -131,38 +131,34 @@ typedef off_t off_T;
 /*
  * The characters and attributes cached for the screen.
  */
-typedef char_u schar_T;
-typedef unsigned short sattr_T;
-
-/*
- * The characters that are currently on the screen are kept in ScreenLines[].
- * It is a single block of characters, the size of the screen plus one line.
- * The attributes for those characters are kept in ScreenAttrs[].
- *
- * "LineOffset[n]" is the offset from ScreenLines[] for the start of line 'n'.
- * The same value is used for ScreenLinesUC[] and ScreenAttrs[].
- *
- * Note: before the screen is initialized and when out of memory these can be
- * NULL.
- */
+typedef char_u schar_T[(MAX_MCO+1) * 4 + 1];
+typedef int16_t sattr_T;
+
+/// ScreenLines[] contains a copy of the whole screen, as it is currently It is
+/// a single block of screen cells, the size of the screen plus one line.
```

Comment:
I found the previous explanation clearer. The "plus one line" could be explained too.

Reply:
A bit of a sentence is missing yes. But other than that, how is an _incorrect_ description clearer? Both before and after this PR: before `ScreenLines` did not if fact contain the characters on the screen, and now that it does, it is not a "block of characters", rather the paragraph below describes the actual format, including at least trying to explain double-width chars, which the previous "explanation" completely ignored.

I wrote a paragraph about "plus one" elsewhere, I will move it here.

Reply:
more precisely the sentence "a copy of the whole screen, as it is currently" the notion of current screen could be understood as what is displayed, which you say it is not.

Reply:
Well, I called it a copy because it is a copy, in the sense it is not essential to displaying the state that it contains, (unlike say in `gvim`, where it is directly used to respond to damage events from the windowing system). It is rather used to remember what we drew until next redraw, so the next redraw can avoid resending cells that are identical.